### DISCUSSION TITLE
Github somehow cannot access the name of the programming language name automatically

### DISCUSSION THREAD

**Discussion by felixoder** [Cyfrin/foundry-full-course-cu#2601] [Q&A]
So the problem is not too serious [just asking for knowledge]
So I had created a simple .sol project on remix and added that to my gh 

[Just added the ss because of giving seamless proofs sry]

![image](https://github.com/user-attachments/assets/e1b54143-d779-4409-927a-41853a444bfc)
now The problem I was getting that github did not able to fetch the language [pl] name automatically which that should do [and I did not had seen any miss ever] 

But after pushing the code I cannot be able to view the Pl name somehow. I searched everywhere what is the solution but it persists. I had recreated the repo but the problem persists.

But today I have update the repo with a .gitattributes file and just added the pl name manually 

```
*.sol linguist-language=Solidity
```

and It works. 

Now my question is what was the problem was it just a fault of my repo structure or something [may be becz it is solidity]

Now it works btw 
![image](https://github.com/user-attachments/assets/76ee92d4-ceb3-40e2-bc50-9f36a5b5ae7b)

**Comment by EngrPips** [comment]
Honestly I am unsure what the problem could be but I am looking forward to the response of someone who knows.
